src/Aspire.Hosting/Dcp/Process/ProcessUtil.cs

```diff
@@ -108,6 +93,21 @@ public static (Task<ProcessResult>, IAsyncDisposable) Run(ProcessSpec processSpe
             process.BeginOutputReadLine();
             process.BeginErrorReadLine();
             processSpec.OnStart?.Invoke(process.Id);
+
+            process.WaitForExitAsync().ContinueWith(t =>
+            {
+                startupComplete.Wait();
+
+                if (processSpec.ThrowOnNonZeroReturnCode && process.ExitCode != 0)
+                {
+                    processLifetimeTcs.TrySetException(new InvalidOperationException(
+                        $"Command {processSpec.ExecutablePath} {processSpec.Arguments} returned non-zero exit code {process.ExitCode}"));
+                }
+                else
+                {
+                    processLifetimeTcs.TrySetResult(new ProcessResult(process.ExitCode));
+                }
+            }, TaskScheduler.Current).ConfigureAwait(false);
```

Comment:
I'm curious - why are we using`TaskScheduler.Current` here?

Reply:
There is a code analysis tool that requires specifying the task scheduler parameter.

<img width="1085" alt="image" src="https://github.com/dotnet/aspire/assets/4603206/16ce95c9-d8ad-4636-807d-0c2275a64269">

Reply:
I could use `TaskScheduler.Default` if that's preferred.

Reply:
I was just curious.